File: mobile/lib/models/product.dart
class Product {
  final int id;
  final String sku;
  final String name;
  final String unit;
  final double sellingPrice;
  final int stockQuantity;
  final String? imageUrl;
  final int? categoryId;
  final String? categoryName;

  Product({
    required this.id,
    required this.sku,
    required this.name,
    required this.unit,
    required this.sellingPrice,
    required this.stockQuantity,
    this.imageUrl,
    this.categoryId,
    this.categoryName,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    final category = json['category'] as Map<String, dynamic>?;
    return Product(
      id: json['id'],
      sku: json['sku'],
      name: json['name'],
      unit: json['unit'] ?? 'pcs',
      sellingPrice: double.parse(json['selling_price'].toString()),
      stockQuantity: json['stock_quantity'] ?? 0,
      imageUrl: json['image_url'],
      categoryId: category?['id'],
      categoryName: category?['name'],
    );
  }
}

File: backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Support\Facades\Storage;

class Product extends Model
{
    protected $fillable = [
        'sku',
        'item_code',
        'barcode',
        'category_id',
        'supplier_id',
        'name',
        'unit',
        'cost_price',
        'selling_price',
        'stock_quantity',
        'low_stock_alert',
        'expiry_date',
        'image_path',
        'is_active',
    ];

    protected $appends = ['image_url'];

    protected function casts(): array
    {
        return [
            'cost_price' => 'decimal:2',
            'selling_price' => 'decimal:2',
            'expiry_date' => 'date',
            'is_active' => 'boolean',
        ];
    }

    public function getImageUrlAttribute(): ?string
    {
        return $this->image_path ? Storage::disk('public')->url($this->image_path) : null;
    }

    public function category(): BelongsTo
    {
        return $this->belongsTo(Category::class);
    }

    public function supplier(): BelongsTo
    {
        return $this->belongsTo(Supplier::class);
    }

    public function saleItems(): HasMany
    {
        return $this->hasMany(SaleItem::class);
    }

    public function stockMovements(): HasMany
    {
        return $this->hasMany(StockMovement::class);
    }

    public function isLowStock(): bool
    {
        return $this->stock_quantity <= $this->low_stock_alert;
    }
}
